src/V3SchedVirtIface.cpp

```diff
@@ -141,130 +110,67 @@ class VirtIfaceVisitor final : public VNVisitor {
-            && nodep->exists([](AstNode* nodep) { return nodep->isTimingControl(); })) {
-            m_trigAssignp = nullptr;
-            m_trigAssignIfacep = nullptr;
-            m_trigAssignMemberVarp = nullptr;
+        VL_RESTORER(m_curStmt);
+        m_curStmt = nodep;
+        iterateChildren(nodep);
+    }
+    void visit(AstVarRef* const nodep) override {
+        if (!m_curStmt) return;
+        if (nodep->access().isReadOnly()) return;
+        AstIface* ifacep = nullptr;
+        AstVar* memberVarp = nullptr;
+        if (AstIfaceRefDType* const dtypep = VN_CAST(nodep->varp()->dtypep(), IfaceRefDType)) {
+            if (dtypep->isVirtual()) {
+                if (AstMemberSel* const memberSelp = VN_CAST(nodep->firstAbovep(), MemberSel)) {
+                    // Extract the member varp from the MemberSel node
+                    memberVarp = memberSelp->varp();
+                    ifacep = dtypep->ifacep();
+                }
+            }
+        } else if ((ifacep = nodep->varp()->sensIfacep())) {
+            memberVarp = nodep->varp();
+        } else if (VN_IS(nodep->backp(), AssignW)) {
+            memberVarp = nodep->varScopep()->varp();
+            ifacep = VN_CAST(memberVarp->backp(), Iface);
         }
-        FileLine* const flp = nodep->fileline();
 
-        foreachWrittenVirtIfaceMember(nodep, [&](AstVarRef*, AstIface* ifacep,
-                                                 AstVar* memberVarp) {
-            if (ifacep != m_trigAssignIfacep || memberVarp != m_trigAssignMemberVarp) {
-                // Write to different interface member than before - need new trigger assignment
-                m_trigAssignIfacep = ifacep;
-                m_trigAssignMemberVarp = memberVarp;
-                m_trigAssignp = nullptr;
-            }
-            if (!m_trigAssignp) {
-                m_trigAssignp
-                    = new AstAssign{flp, createVirtIfaceMemberTriggerRefp(flp, ifacep, memberVarp),
-                                    new AstConst{flp, AstConst::BitTrue{}}};
-                nodep->addNextHere(m_trigAssignp);
-            }
-        });
-        // Fallback to whole-interface tracking if no member-specific assignments found
-        if (!m_trigAssignp) {
-            foreachWrittenVirtIface(nodep, [&](AstVarRef*, AstIface* ifacep) {
-                if (ifacep != m_trigAssignIfacep) {
-                    m_trigAssignIfacep = ifacep;
-                    m_trigAssignMemberVarp = nullptr;
-                    m_trigAssignp = nullptr;
-                }
-                if (!m_trigAssignp) {
-                    m_trigAssignp = new AstAssign{flp, createVirtIfaceTriggerRefp(flp, ifacep),
-                                                  new AstConst{flp, AstConst::BitTrue{}}};
-                    nodep->addNextHere(m_trigAssignp);
-                }
-            });
+        if (ifacep && memberVarp) {
+            FileLine* const flp = nodep->fileline();
+            m_curStmt->addNextHere(
```

Comment:
If you use `m_currStmtp->assHereThisAsNext` to insert before the current statement, it shoujld then naturally work with AstNodeIf and AstLoopTest conditions.